lib/logstash/filters/resolv_patch.rb

```diff
@@ -0,0 +1,29 @@
+require "resolv"
+
+# ref: https://github.com/logstash-plugins/logstash-filter-dns/issues/40
+#
+# JRuby versions prior to 9.1.16.0 have a bug which crashes IP address
+# the resolver after 64k unique IP addresses resolutions.
+
+version = JRUBY_VERSION.split(".").map(&:to_i)
+
+if version[0] == 9 && version[1] == 1 && version[2] < 16
+  class Resolv
```

Comment:
It'd be cool to document where this code comes from: https://github.com/jruby/jruby/blob/9.1.16.0/lib/ruby/stdlib/resolv.rb#L775